java/src/main/java/ai/rapids/cudf/Table.java

```diff
@@ -1727,52 +1745,60 @@ public Table aggregate(Aggregate... aggregates) {
       int keysLength = operation.indices.length;
       int totalOps = 0;
       for (int outputIndex = 0; outputIndex < aggregates.length; outputIndex++) {
-        Aggregate agg = aggregates[outputIndex];
-        ColumnOps ops = groupedOps.computeIfAbsent(agg.getIndex(), (idx) -> new ColumnOps());
-        totalOps += ops.add(agg.getOp(), outputIndex + keysLength);
+        AggregationOnColumn agg = aggregates[outputIndex];
+        ColumnOps ops = groupedOps.computeIfAbsent(agg.getColumnIndex(), (idx) -> new ColumnOps());
+        totalOps += ops.add(agg, outputIndex + keysLength);
       }
       int[] aggColumnIndexes = new int[totalOps];
-      int[] aggOperationIds = new int[totalOps];
-      int opIndex = 0;
-      for (Map.Entry<Integer, ColumnOps> entry: groupedOps.entrySet()) {
-        int columnIndex = entry.getKey();
-        for (AggregateOp operation: entry.getValue().operations()) {
-          aggColumnIndexes[opIndex] = columnIndex;
-          aggOperationIds[opIndex] = operation.nativeId;
-          opIndex++;
-        }
-      }
-      assert opIndex == totalOps: opIndex + " == " + totalOps;
-
-      Table aggregate;
-      aggregate = new Table(groupByAggregate(
-          operation.table.nativeHandle,
-          operation.indices,
-          aggColumnIndexes,
-          aggOperationIds,
-          groupByOptions.getIgnoreNullKeys()));
+      long[] aggOperationInstances = new long[totalOps];
       try {
-        // prepare the final table
-        ColumnVector[] finalCols = new ColumnVector[keysLength + aggregates.length];
-
-        // get the key columns
-        for (int aggIndex = 0; aggIndex < keysLength; aggIndex++) {
-          finalCols[aggIndex] = aggregate.getColumn(aggIndex);
+        int opIndex = 0;
+        for (Map.Entry<Integer, ColumnOps> entry: groupedOps.entrySet()) {
+          int columnIndex = entry.getKey();
+          for (Aggregation operation: entry.getValue().operations()) {
+            aggColumnIndexes[opIndex] = columnIndex;
+            aggOperationInstances[opIndex] = operation.createNativeInstance();
+            opIndex++;
+          }
         }
+        assert opIndex == totalOps : opIndex + " == " + totalOps;
+
+        Table aggregate;
+        aggregate = new Table(groupByAggregate(
+                operation.table.nativeHandle,
```

Comment:
Nit: indentation seems off here?  Expected 4-space indent on a continuation.

Reply:
Looks like IntelliJ dropped my settings again.  I re-imported them and it looks like that fixed it.